lighthouse-assert/index.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import {Assert, IExpectation, IResult} from './assert/assert';
+import {Logger} from './logger';
+import {DefaultReporter, IReporter} from './reporter/reporter';
+
+export class LighthouseAssert {
+  private assertInstance: Assert;
+  private logger: IReporter;
+
+  constructor(reporter?: IReporter) {
+    reporter = reporter || new DefaultReporter();
+    this.logger = new Logger(reporter);
+  }
+
+  assert(results: Array<IResult>, expectations: Array<IExpectation>) {
+    this.assertInstance = new Assert(results, expectations);
+    this.assertInstance.collate();
```

Comment:
the usage pattern of this feels a bit weird to me, is this modeling something pre-existing that I'm not aware of?

I ask because I would expect collate to return some object that contains the results found rather than it modifying the object and needing to call other methods to find out what happened.

Reply:
No, it's not smth pre-existing :) , I just wanted to compare results not during instantiating class but with other action, that's why I added `collate` method. Another idea was to reuse `collatedResults` in future e.g. for comparing like `equal`, but right now I realized that was not a good idea...